src/antelope/stores/history.ts

```diff
@@ -179,42 +296,93 @@ export const useHistoryStore = defineStore(store_name, {
 
                 if (!isFailed) {
                     if (+tx.value) {
-                        const valueInFiatBn = convertCurrency(BigNumber.from(tx.value), WEI_PRECISION, 2, tlosInUsd);
+                        const valueInFiatBn = convertCurrency(BigNumber.from(tx.value), systemTokenDecimals, 2, tlosInUsd);
                         const valueInFiat = +formatUnits(valueInFiatBn, 2);
 
                         if (tx.from?.toLowerCase() === userAddressLower) {
                             valuesOut.push({
-                                amount: +formatUnits(tx.value, WEI_PRECISION),
+                                amount: +formatUnits(tx.value, systemTokenDecimals),
                                 symbol: chainSettings.getSystemToken().symbol,
                                 fiatValue: valueInFiat,
                             });
                         }
                         if (tx.to?.toLowerCase() === userAddressLower) {
                             valuesIn.push({
-                                amount: +formatUnits(tx.value, WEI_PRECISION),
+                                amount: +formatUnits(tx.value, systemTokenDecimals),
                                 symbol: chainSettings.getSystemToken().symbol,
                                 fiatValue: valueInFiat,
                             });
                         }
                     }
 
-                    for (const tokenXfer of transfers) {
+                    const nftTransfersInTx = this.getEvmNftTransfersForTransaction(label, tx.hash);
+
+                    if (nftTransfersInTx.length > 0) {
+                        // there is at least 1 NFT transfer in this transaction,
+                        // so we need to fetch the NFT details for each transfer in this transaction
+                        const nftDetailList = await Promise.all(
+                            nftTransfersInTx.map(
+                                transfer => nftStore.fetchNftDetails(
+                                    label,
+                                    transfer.contract,
+                                    transfer.id ?? '',
+                                    transfer.type.toUpperCase() as NftTokenInterface,
+                                ),
+                            ),
+                        );
+
+                        nftDetailList.forEach((nftDetails, index) => {
+                            if (nftDetails) {
+                                let nftMediaType: 'image' | 'video' | 'audio' | 'unknown' = 'unknown';
+
+                                if (nftDetails.audioSrc) {
+                                    nftMediaType = 'audio';
+                                } else if (nftDetails.videoSrc) {
+                                    nftMediaType = 'video';
+                                } else if (nftDetails.imageSrcFull) {
+                                    nftMediaType = 'image';
+                                }
```

Comment:
You can simply take the media type from the nftDirectly this way:
`const nftMediaType = nftDetails.item.type;`
The only problem would be that the 'unknown' value is not part of the current list of values. You can rename the NFTSourceTypes.NONE to NFTSourceTypes.UNKNOWN (which I think is the best) or you can adapt your code to use the 'none' value instead of 'unknown'